pkg/service/replay/replay.go

```diff
@@ -1153,39 +1168,67 @@ func (r *Replayer) printSummary(_ context.Context, _ bool) {
 				utils.LogError(r.logger, err, "failed to print test run summary")
 				return
 			}
-			if _, err := pp.Printf("\n\tTest Suite Name\t\tTotal Test\tPassed\t\tFailed\t\tIgnored\t\tTime Taken\t\n"); err != nil {
-				utils.LogError(r.logger, err, "failed to print test suite summary")
-				return
-			}
 		} else {
 			if _, err := pp.Printf("\n <=========================================> \n  COMPLETE TESTRUN SUMMARY. \n\tTotal tests: %s\n"+"\tTotal test passed: %s\n"+"\tTotal test failed: %s\n"+"\tTotal time taken: %s\n", totalTests, totalTestPassed, totalTestFailed, totalTestTimeTakenStr); err != nil {
 				utils.LogError(r.logger, err, "failed to print test run summary")
 				return
 			}
-			if _, err := pp.Printf("\n\tTest Suite Name\t\tTotal Test\tPassed\t\tFailed\t\tTime Taken\t\n"); err != nil {
-				utils.LogError(r.logger, err, "failed to print test suite summary")
-				return
-			}
 		}
+
+		header := "\n\tTest Suite Name\t\tTotal Test\tPassed\t\tFailed"
+		if totalTestIgnored > 0 {
+			header += "\t\tIgnored"
+		}
+		header += "\t\tTime Taken"
+		if totalTestFailed > 0 && !r.config.Test.MustPass {
+			header += "\tFailed Testcase Names"
```

Comment:
Add comment why did you use mustPass here and also the name of the header should just be "Failed Testcases"